src/curdleproofs.rs

```diff
@@ -54,8 +53,8 @@ pub fn generate_crs(ell: usize) -> CurdleproofsCrs {
     let crs_H = G1Projective::rand(&mut rng);
     let crs_G_t = G1Projective::rand(&mut rng);
     let crs_G_u = G1Projective::rand(&mut rng);
-    let crs_G_sum: G1Affine = crs_G_vec.iter().sum();
-    let crs_H_sum: G1Affine = crs_H_vec.iter().sum();
+    let crs_G_sum: G1Affine = sum_affine_points(&crs_G_vec);
```

Comment:
The sum method has become quite verbose, so I put it in a utils file. It needs to convert to Projective, sum and then convert to Affine. This is likely what the other implementation was doing, or it may have been doing mixed addition, you can probably experiment here to see which is faster